cleanenv.go

```diff
@@ -459,15 +460,27 @@ func readEnvVars(cfg interface{}, update bool) error {
 func parseValue(field reflect.Value, value, sep string, layout *string) error {
 	// TODO: simplify recursion
 
+	valueType := field.Type()
+
+	// look for supported struct parser
+	if structParser, found := validStructs[valueType]; found {
+		return structParser(&field, value, layout)
+	}
+
 	if field.CanInterface() {
+		if ctp, ok := field.Addr().Interface().(encoding.TextUnmarshaler); ok {
+			return ctp.UnmarshalText([]byte(value))
+		} else if ct, ok := field.Interface().(encoding.TextUnmarshaler); ok {
+			return ct.UnmarshalText([]byte(value))
+		}
+
 		if cs, ok := field.Interface().(Setter); ok {
 			return cs.SetValue(value)
 		} else if csp, ok := field.Addr().Interface().(Setter); ok {
 			return csp.SetValue(value)
 		}
```

Comment:
I think the code looks better if it is consistent. Like if we do similar things in a similar order, like check `field.Interface()` first and then `field.Addr().Interface()`. Not a big deal, but since it's a public code I want to keep it looking nice.

Reply:
agree, done ✅